File: CT/WTC/2.py
def solution(s, op):
    answer = []
    for i in range(1, len(s)):
        if op == "+":
            result = int(s[:i]) + int(s[i:])
        elif op == "-":
            result = int(s[:i]) - int(s[i:])
        else:
            result = int(s[:i]) * int(s[i:])
        answer.append(result)

    return answer

File: CT/Dev/2.java
import java.util.Stack;

class Solution {
    public int[] solution(int[] deposit) {
        Stack<Integer> history = new Stack<>();
        
        for (int money:deposit) {
            if (money > 0) {
                history.push(money);
            } else {
                int sum = 0; // 현재까지 출금한 금액
                int abs = -(money);
                while (sum < abs) {
                    int diff = abs - sum;
                    int last = history.pop();
                    if (last > diff) {
                        history.push(last - diff);
                    }
                    sum += last;
                }
            }
        }
        
        int[] answer = history.stream().mapToInt(i->i).toArray();
        
        return answer;
    }
}